client/payments/data/types.ts

```diff
@@ -0,0 +1,32 @@
+export type PaymentGateway = {
+	id: string;
+	title: string;
+	description: string;
+	order: number | string;
+	enabled: boolean;
+	method_title: string;
+	method_description: string;
+	method_supports: Array< string >;
+	settings: Record< string, Record< string, string > >;
```

Comment:
As mentioned in the `test/stub.ts` file, I think we should create a type of the setting definition.